util/dvsim/Launcher.py

```diff
@@ -272,6 +298,7 @@ def _find_patterns(patterns, line):
                     if _find_patterns(self.deploy.fail_patterns, line):
                         # If failed, then nothing else to do. Just return.
                         # Privide some extra lines for context.
+                        _get_runtimes(lines, self.deploy.sim_cfg.tool)
```

Comment:
Is this a mistake?

Reply:
Oh good catch, I should update the comment! If we do not call this `_get_runtimes` function here, we won't be able to parse runtimes when the simulation failed.

Reply:
Wouldn't line 312 compute the runtimes regardless of whether the sim passed or failed, given its outside the line 295 `if` clause?

Reply:
If I understand it correctly - line 302 will return the function and won't execute line 312 rightt?

Reply:
Aah yes. In that case, invoke it once before line 295?

Reply:
yeah! That would work! Thanks for the suggestion.